### PR TITLE
Update storage in set_client_options

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Update storage in set_client_options

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Cli and updated test

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by thibault-martinez — DISMISSED** [review]

**Review by kwek20 — DISMISSED** [review]

**Review by DaughterOfMars — DISMISSED** [review]

File: sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs [hunk 1]
```diff
@@ -148,6 +138,17 @@ impl Wallet {
 
         stronghold_password.zeroize();
 
+        // drop secret manager, otherwise we get a deadlock in set_client_options() (there inside of save_wallet_data())
+        drop(secret_manager);
```
Comment (DaughterOfMars) [inline sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs:144]: What is the reason for the deadlock?

Comment (DaughterOfMars) [reply sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs:144]: Oh, we're holding the write lock above, nvm.

Comment (Thoralf-M) [reply sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs:144]: In restore_backup() we write() on the secret_manager to be able to replace it with the restored one and in save_wallet_data we have `secret_manager.read().await;` to convert the SecreManager to its Dto

File: sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs [hunk 2]
```diff
@@ -175,30 +176,9 @@ impl Wallet {
...
-                .with_client_options(self.client_options().await)
-                .with_coin_type(self.coin_type.load(Ordering::Relaxed));
-            // drop secret manager, otherwise we get a deadlock in save_wallet_data
-            drop(secret_manager);
-            self.storage_manager
-                .read()
-                .await
-                .save_wallet_data(&wallet_builder)
-                .await?;
-            // also save account to db
```
Comment (thibault-martinez) [inline sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs]: Why is this removed?

Comment (Thoralf-M) [reply sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs]: Because it's saved when we set the client options

Comment (thibault-martinez) [reply sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs]: Ok but this was always done, while `set_client_options` is conditionally done

Comment (Thoralf-M) [reply sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs]: Oh right, reverted

**Review by DaughterOfMars — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]
